<a href="https://colab.research.google.com/github/gaso/tipologia_heart-attack-analysis/blob/main/analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<div style="width: 100%; clear: both;">
<div style="float: left; width: 50%;">
<img src="http://www.uoc.edu/portal/_resources/common/imatges/marca_UOC/UOC_Masterbrand.jpg", align="left">
</div>
<div style="float: right; width: 50%;">
<p style="margin: 0; padding-top: 22px; text-align:right;">Tipología y ciclo de vida de los datos</p>
<p style="margin: 0; text-align:right;">2023-1 · Máster universitario en Ciencia de datos (Data science)</p>
<p style="margin: 0; text-align:right; padding-button: 100px;">Estudios de Informática, Multimedia y Telecomunicación</p>
</div>
</div>
<div style="width:100%;">&nbsp;</div>

Cargue de Librerias

In [ ]:
install.packages('ggplot2')

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



# 1. Descripción del dataset.

## 1.1 Carga del dataset

In [1]:
path= 'https://raw.githubusercontent.com/gaso/tipologia_heart-attack-analysis/main/dataset/heart.csv'
data <- read.csv(path, row.names=NULL)

str(data)

'data.frame':	303 obs. of  14 variables:
 $ age     : int  63 37 41 56 57 57 56 44 52 57 ...
 $ sex     : int  1 1 0 1 0 1 0 1 1 1 ...
 $ cp      : int  3 2 1 1 0 0 1 1 2 2 ...
 $ trtbps  : int  145 130 130 120 120 140 140 120 172 150 ...
 $ chol    : int  233 250 204 236 354 192 294 263 199 168 ...
 $ fbs     : int  1 0 0 0 0 0 0 0 1 0 ...
 $ restecg : int  0 1 0 1 1 1 0 1 1 1 ...
 $ thalachh: int  150 187 172 178 163 148 153 173 162 174 ...
 $ exng    : int  0 0 0 0 1 0 0 0 0 0 ...
 $ oldpeak : num  2.3 3.5 1.4 0.8 0.6 0.4 1.3 0 0.5 1.6 ...
 $ slp     : int  0 0 2 2 2 1 1 2 2 2 ...
 $ caa     : int  0 0 0 0 0 0 0 0 0 0 ...
 $ thall   : int  1 2 2 2 2 1 2 3 3 2 ...
 $ output  : int  1 1 1 1 1 1 1 1 1 1 ...


El conjunto de datos contiene **303** observaciones y **14** variables. 

**age:** La edad del paciente (variable numérica).

**sex:** El sexo del paciente (1: masculino, 0: femenino) (variable numérica).

**cp:** Tipo de dolor en el pecho (variable numérica).

**trtbps:** Presión arterial en reposo (variable numérica).

**chol:** Nivel de colesterol en mg/dL (variable numérica).

**fbs:** Nivel de azúcar en la sangre en ayunas (1: superior a 120 mg/dL, 0: inferior a 120 mg/dL) (variable numérica).

**restecg:** Resultados del electrocardiograma en reposo (variable numérica).

**thalachh:** Frecuencia cardíaca máxima alcanzada (variable numérica).

**exng:** Angina inducida por ejercicio (1: sí, 0: no) (variable numérica).

**oldpeak:** Depresión del ST inducida por el ejercicio en relación con el descanso (variable numérica).

**slp: **Pendiente del segmento ST de ejercicio (variable numérica).

**caa:** Número de vasos principales coloreados por fluoroscopia (variable numérica).

**thall:** Resultados de las pruebas de estrés con talio (variable numérica).

**output:** Variable de salida, 1 si la persona tiene enfermedad cardíaca y 0 si no la tiene (variable numérica).

Resumen de datos que contiene el dataset

In [4]:
head(data)

age sex cp trtbps chol fbs restecg thalachh exng oldpeak slp caa thall output
1 63  1   3  145    233  1   0       150      0    2.3     0   0   1     1     
2 37  1   2  130    250  0   1       187      0    3.5     0   0   2     1     
3 41  0   1  130    204  0   0       172      0    1.4     2   0   2     1     
4 56  1   1  120    236  0   1       178      0    0.8     2   0   2     1     
5 57  0   0  120    354  0   1       163      1    0.6     2   0   2     1     
6 57  1   0  140    192  0   1       148      0    0.4     1   0   1     1

In [3]:
summary(data)

      age             sex               cp            trtbps     
 Min.   :29.00   Min.   :0.0000   Min.   :0.000   Min.   : 94.0  
 1st Qu.:47.50   1st Qu.:0.0000   1st Qu.:0.000   1st Qu.:120.0  
 Median :55.00   Median :1.0000   Median :1.000   Median :130.0  
 Mean   :54.37   Mean   :0.6832   Mean   :0.967   Mean   :131.6  
 3rd Qu.:61.00   3rd Qu.:1.0000   3rd Qu.:2.000   3rd Qu.:140.0  
 Max.   :77.00   Max.   :1.0000   Max.   :3.000   Max.   :200.0  
      chol            fbs            restecg          thalachh    
 Min.   :126.0   Min.   :0.0000   Min.   :0.0000   Min.   : 71.0  
 1st Qu.:211.0   1st Qu.:0.0000   1st Qu.:0.0000   1st Qu.:133.5  
 Median :240.0   Median :0.0000   Median :1.0000   Median :153.0  
 Mean   :246.3   Mean   :0.1485   Mean   :0.5281   Mean   :149.6  
 3rd Qu.:274.5   3rd Qu.:0.0000   3rd Qu.:1.0000   3rd Qu.:166.0  
 Max.   :564.0   Max.   :1.0000   Max.   :2.0000   Max.   :202.0  
      exng           oldpeak          slp             caa        
 Mi

Comprobar valores nulos en conjunto de datos

In [2]:
colSums(is.na(data))

age      sex       cp   trtbps     chol      fbs  restecg thalachh 
       0        0        0        0        0        0        0        0 
    exng  oldpeak      slp      caa    thall   output 
       0        0        0        0        0        0

Comprobar duplicados

In [10]:
duplicated_rows <- data[duplicated(data), ]
duplicated_rows

age sex cp trtbps chol fbs restecg thalachh exng oldpeak slp caa thall
165 38  1   2  138    175  0   1       173      0    0       2   4   2    
    output
165 1